devices/webcam/hand_tracking_utility.py

```diff
@@ -0,0 +1,600 @@
+
+class PoseFrame(lg.Message):
+    """
+    Represents a single frame of a webcam's stream along with its hand tracking estimations
+
+    @attributes:
+        frame: a (W, H, 3) shape array of the webcam's stream at its default resolution, in the blue/green/red color space
+        landmarks: List[NormalizedLandmarkList], one NormalizedLandmarkList object for each detected hand
+    
+    NormalizedLandmarkList:
+        @attributes:
+            landmark: RepeatedCompositeContainer, list-like container with up to 21 hand pose landmarks with their normalized (X, Y, Z) coordinates as attributes
+    """
+
+    frame: np.ndarray
+    landmarks: List[NormalizedLandmarkList]
+
+class PoseEstimationConfig(lg.Config):
+    """
+    Configuration for MediaPipe's hand pose esitmation graph
+    See https://google.github.io/mediapipe/solutions/hands.html
+
+    @attributes:
+        model_complexity: int
+        min_detection_confidence: float
+        min_tracking_confidence: float
+    """
+
+    model_complexity: int
+    min_detection_confidence: float
+    min_tracking_confidence: float
+
+class PoseEstimationState(lg.State):
+    """
+    State object for hand pose estimation node
+    See https://google.github.io/mediapipe/solutions/hands.html
+
+    @attributes:
+        hands: mediapipe.python.solutions.hands.Hands
+    """
+
+    hands: Optional[HandsType.Hands] = None
+
+class PoseEstimation(lg.Node):
+    """
+    PoseEstimation node, runs VideoFrame streams through MediaPipe's Hands graph to generate pose data
+
+    @topics:
+        INPUT: VideoFrame
+        OUTPUT: PoseFrame
+    
+    @attributes:
+        config: PoseEstimationConfig
+        state: PoseEstimationState
+    """
+
+    INPUT = lg.Topic(VideoFrame)
+    OUTPUT = lg.Topic(PoseFrame)
+    config: PoseEstimationConfig
+    state: PoseEstimationState
+
+    @lg.subscriber(INPUT)
+    @lg.publisher(OUTPUT)
+    async def processFrame(self, message: VideoFrame) -> lg.AsyncPublisher:
+        results: NamedTuple = self.state.hands.process(cv2.cvtColor(message.frame, cv2.COLOR_BGR2RGB))
+        landmarks = results.multi_hand_landmarks
+        if landmarks is None:
+            landmarks = []
+        yield self.OUTPUT, PoseFrame(landmarks = landmarks, frame = message.frame)
+
+    def setup(self):
+        self.state.hands = mp_hands.Hands(
+            model_complexity = self.config.model_complexity,
+            min_detection_confidence = self.config.min_detection_confidence,
+            min_tracking_confidence = self.config.min_tracking_confidence)
+
+class LoggingConfig(lg.Config):
+    """
+    Configuration for the data logging node
+
+    @attributes:
+        log_images: bool, whether or not to log images to file
+        log_poses: bool, whether or not to log pose data to file
+        file_path: str, the path of the log file
+    """
+
+    log_images: bool
```

Comment:
Default values can be added in the config file.

Reply:
After the latest series of commits, we no longer make use of a `lg.Config` object for the graph. Is this acceptable?